ffmpeg/_run.py

```diff
@@ -57,8 +57,8 @@ def _get_input_args(input_node):
 def _get_filter_spec(node, outgoing_edge_map, stream_name_map):
     incoming_edges = node.incoming_edges
     outgoing_edges = get_outgoing_edges(node, outgoing_edge_map)
-    inputs = [stream_name_map[edge.upstream_node, edge.upstream_label] for edge in incoming_edges]
-    outputs = [stream_name_map[edge.upstream_node, edge.upstream_label] for edge in outgoing_edges]
+    inputs = ["[{}{}]".format(stream_name_map[edge.upstream_node, edge.upstream_label], "" if not edge.upstream_selector else ":{}".format(edge.upstream_selector)) for edge in incoming_edges]
```

Comment:
This line has gotten too complicated and is basically unreadable. Can we split it out a bit?
e.g.
```
def _get_filter_spec(node, outgoing_edge_map, stream_name_map):
    def format_stream_name(edge):
        prefix = stream_name_map[edge.upstream_node, edge.upstream_label]
        if not edge.upstream_selector
            suffix = ""
        else:
            suffix = ":{}".format(edge.upstream_selector)
        return "[{}{}]".format(prefix, suffix)

    incoming_edges = node.incoming_edges
    outgoing_edges = get_outgoing_edges(node, outgoing_edge_map)
    inputs = [stream_name_map[edge.upstream_node, edge.upstream_label] for edge in incoming_edges]
    outputs = [stream_name_map[edge.upstream_node, edge.upstream_label] for edge in outgoing_edges]
    inputs = map(format_stream_name, incoming_edges)
    inputs = [format_stream_name(edge) for edge in incoming_edges]
```

Reply:
Sure.

Reply:
Done 7fb7026e